spec/codelog/cli_spec.rb

```diff
@@ -8,16 +8,30 @@
     end
   end
 
-  context 'with generic commands' do
-    GENERIC_COMMANDS = ['setup', 'new'].freeze
+  describe '#new' do
+    before :each do
+      allow(Codelog::Command::New).to receive(:run)
+    end
 
-    GENERIC_COMMANDS.each do |command|
-      it "calls the #{command} command" do
-        command_class = Module.const_get "Codelog::Command::#{command.capitalize}"
-        allow(command_class).to receive(:run)
-        subject.send(command)
-        expect(command_class).to have_received(:run)
-      end
+    it 'calls the new command' do
+      subject.new
+      expect(Codelog::Command::New).to have_received(:run)
+    end
+
+    it 'pass the change name to the command' do
+      subject.new('test_name')
+      expect(Codelog::Command::New).to have_received(:run).with('test_name', {})
```

Comment:
What does `{}` represents? If those are the passed options wouldn't it be better to make it more explicit or even create a test for it?

Reply:
I believe `Thor` is responsible for the management of this `options` var, so I don't think its worth a test.
But naming this empty hash `default_options` would surely make the test more readable 😃